### PR TITLE
Add more support for nvcc

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsvu — APPROVED** [review]

File: src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp [hunk 1]
```diff
@@ -29,8 +29,8 @@ struct RayInterpolant {
     return apply_matrices(interpolation_matrices, values, mesh.extents())[0];
   }
 
-  const DataVector& values{};
-  const Mesh<1>& mesh{};
+  const DataVector& values;
+  const Mesh<1>& mesh;
```
Comment (knelli2) [inline src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp:33]: [optional] Either
1. NOLINT these
2. Make the mesh not a ref (it's computed using `slice_through` anyways) and the DataVector non-owning
3. Use pointers

**Review by knelli2 — APPROVED** [review]

File: src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp (hunk 1 above)
Comment (nilsdeppe) [reply src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp:33]: @nilsvu I think this is code for ID or something? I don't know what it's supposed to do so could you take over addressing this? I remember looking at the code and the current use-case is technically fine.

Comment (nilsvu) [reply src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp:33]: This is for BNS ID. Yes I can address this in a separate PR.

Comment (knelli2) [reply src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp:33]: Good enough for me

**Pull request merged**